src/app/src/App.js

```diff
@@ -8,13 +8,42 @@ import Header from './Header';
 import Footer from './Footer';
 import SensorOverview from './SensorOverview';
 
+import { setSensorData } from './app.actions';
+import { POLLING_INTERVAL, SENSORS } from './constants';
+
+import {
+    makeRiverGaugeRequest,
+    parseRiverGaugeApiData,
+    parseRiverGaugeCsvData,
+} from './sensorUtils';
+
 class App extends Component {
+    componentDidMount() {
+        // Poll for new sensor data immediately and on a timed cycle thereafter
+        setInterval(function pollData() {
+            (async () => {
```

Comment:
Generally a fan of the `async/await` for cases like this, but would it be possible to migrate this code to an action? 

Also, I'm not sure if this is possible or if it even makes sense with `async/await` (@kellyi may have tried this at some point), but in the past with async redux-thunk actions, we had two actions (`START_FOO`/`COMPLETE_FOO`/`FAIL_FOO`). Is that pattern something we want to try and keep with `async/await`? Here are some examples of what I mean: https://github.com/azavea/american-water-tank-management/blob/develop/src/app/js/actions.flights.js#L12

Reply:
Yeah, async/await will work with the typical start/fail/complete pattern we use for making API requests via Redux, but I think it'd involve using a `try/catch` block rather than the usual Promise chaining syntax inside the Redux-thunk action.

Something like:

```
try {
   const data = await axios.get('whatever');
   return dispatch(completeFetchSensorData(data));
} catch ... {
   return dispatch(failFetchSensorData());
}
```

...would work.

Seems like the entirety of the function implementation here could just be put into a Redux action, leaving just the `setInterval`, the function call, and the timing setting here. Additionally: if it's possible that this App component could ever unmount in the normal course of the application staying in the same session, it'd be worthwhile to add a rhyming `clearInterval` function in `componentWillUnmount` to prevent a memory leak.

One other quick comment: I think `Promise.all` is blocking until all of them have succeeded. I don't know much about how this app is implemented but it might be preferable to make separate sets of Redux actions for each request loop, then store them as three separate things in the app state, since it's conceivable that there'll be times when not all of these requests succeed, but we'd want to show fallback data for the failed cases and real data for the successful ones.

Reply:
Thanks for the suggestions. I'm migrating the logic to `app.actions` and any function in said file results in the mysterious error `TypeError: actions[action].assignTo is not a function`. Ever seen it ?

Reply:
Yeah, that's happening because redux-act's `assignAll` functionality doesn't work with Redux-Thunk actions. I think you'll need to import that action directly from `app.actions` into the component, and then wrap it in dispatch.

https://github.com/azavea/ism-watershed-wellness-snapshot/blob/develop/src/app/src/Map.js#L101 has an example of using the `mapDispatchToProps` function to do that, and there's another one here: https://github.com/open-apparel-registry/open-apparel-registry/blob/develop/src/app/src/components/UserAPITokens.jsx#L173

Reply:
This thread has been helpful. I've moved the action to its own file that exists outside redux-thunk middleware, because of the errors I was encountering a la `assignAll`. Requests to sensors are also made separately, see 660e024

It feels a little messy still that the sensor actions live across two files, but this is the working solution I came to .